src/add_malfunction_flag.R

```diff
@@ -27,24 +19,84 @@ add_malfunction_flag <- function(df, malfunction_records){
+    dplyr::mutate(end_DT = as.POSIXct(end_DT, tz = "MST")) %>%
+    tibble::rowid_to_column()
 
   if (nrow(malfunction_records_filtered > 0)) {
 
+    # Identify known (field crew input) instances of...
+
+    # Sensor drift
+    drift <- malfunction_records_filtered %>%
+      dplyr::filter(grepl("grime|gunk|drift|biofoul|biofilm", notes, ignore.case = TRUE))
+
+    # Sensor burial
+    burial <- malfunction_records_filtered %>%
+      dplyr::filter(grepl("buried|burial|bury|sediment|roots", notes, ignore.case = TRUE))
+
+    # Bad level calibrations (is now built-in to `fix_calibration.R`, SHOULD BE redundant flag)
+    depth_funk <- malfunction_records_filtered %>%
+      dplyr::filter(grepl("improper level calibration", notes, ignore.case = TRUE))
+
+    # Sensors not in water:
+    unsubmerged <- malfunction_records_filtered %>%
+      dplyr::filter(grepl("not submerged", notes, ignore.case = TRUE))
+
+    # All other sensor malfunctions:
+    general_malfunction <- malfunction_records_filtered %>%
+      dplyr::filter(!rowid %in% drift$rowid & !rowid %in% burial$rowid &
+               !rowid %in% depth_funk$rowid & !rowid %in% unsubmerged$rowid)
+
     # make a list of date intervals where sensor was malfunctioning
-    malfunction_interval_list <- map2(
-      .x = malfunction_records_filtered$start_DT,
-      .y = malfunction_records_filtered$end_DT,
-      .f = ~interval(.x, .y, tz = "MST"))
+    drift_interval_list <- map2(
+      .x = drift$start_DT,
+      .y = drift$end_DT,
+      .f = ~lubridate::interval(.x, .y, tz = "MST"))
+
+    burial_interval_list <- map2(
+      .x = burial$start_DT,
+      .y = burial$end_DT,
+      .f = ~lubridate::interval(.x, .y, tz = "MST"))
+
+    depth_interval_list <- map2(
+      .x = depth_funk$start_DT,
+      .y = depth_funk$end_DT,
+      .f = ~lubridate::interval(.x, .y, tz = "MST"))
+
+    unsubmerged_interval_list <- map2(
+      .x = unsubmerged$start_DT,
+      .y = unsubmerged$end_DT,
+      .f = ~lubridate::interval(.x, .y, tz = "MST"))
+
+    general_interval_list <- map2(
+      .x = general_malfunction$start_DT,
+      .y = general_malfunction$end_DT,
+      .f = ~lubridate::interval(.x, .y, tz = "MST"))
 
     # If DT_round is within any of the intervals where a sensor was malfunctioning, flag it
-    df <- df %>%
-      add_flag(DT_round %within% malfunction_interval_list, "sensor malfunction")
+    try(df <- df %>%
+          add_flag(DT_round %within% burial_interval_list, "sonde burial"))
+
+    try(df <- df %>%
+          add_flag(DT_round %within% drift_interval_list, "sensor biofouling"))
+
+    try(df <- df %>%
+          add_flag(DT_round %within% depth_interval_list, "depth calibration malfunction"))
+
+    try(df <- df %>%
+          add_flag(DT_round %within% unsubmerged_interval_list, "sonde unsubmerged"))
+
+    try(df <- df %>%
+          add_flag(DT_round %within% general_interval_list, "sensor malfunction"))
+
```

Comment:
Question: Does this error out unless its wrapped in `try()`?

Reply:
It must have, or I wouldn't have wrapped them in `try()` ... but I honestly don't remember what causes the error. I think it errors if the df doesn't have anything listed in the `_iterval_list` objects...